zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -251,14 +259,27 @@ where
 
         // TODO:
         // We should eventually implement these checks in this function:
-        // - Zebra should stop believing far-future last_seen times from peers (#1871)
         // - Zebra should ignore peers that are older than 3 weeks (part of #1865)
         //   - Zebra should count back 3 weeks from the newest peer timestamp sent
         //     by the other peer, to compensate for clock skew
         // - Zebra should limit the number of addresses it uses from a single Addrs
         //   response (#1869)
 
-        addrs
+        let addrs: Vec<_> = addrs.into_iter().collect();
+        let now = Utc::now();
+
+        let most_recent_reported_seen_time = addrs
+            .iter()
+            .map(|addr| addr.get_last_seen())
+            .max()
+            .unwrap_or(now);
+
+        let offset = now - most_recent_reported_seen_time;
+
+        addrs.into_iter().map(move |mut addr| {
+            addr.offset_last_seen_by(offset);
+            addr
+        })
```

Comment:
We should definitely offset future times so they are at or before the current time.
But I think it's ok to leave past times as they are.

What do you think?
Is there a scenario where believing a past time is an issue?

Reply:
It's kind of minor I guess, but if the times are all in the far past (say, more than a day?) then I think those peers won't be prioritized in the queue, even thought they might be valid (but the reporting peer has a skewed clock).

I figured that by updating all of them, all connected peers have a "fair" chance to get at least of their reported peers attempted "soon" (because the most recent `last_seen` becomes `Utc::now()`).

Reply:
I think fairness and security are different goals though?

The goal of this PR and ticket is to prevent malicious peers dominating the peer set with their own addresses. Tickets like #1875 deal with fairness - but we'll tackle them later.

In general if a peer tells us that all their peers have been down for a while, I think we should believe them.

Reply:
For example, `zcashd` only connects to 8 peers at a time. So it is possible that all its peers happen to have been down for a while. (Or that its network has been down, and we're the first peer to reconnect, and so it is telling us its uptime info on other peers is out of date.)

Reply:
Got it. I'll change it to only update the `addr`s that are in the future.